drivers/hid/hid-multitouch.c

```diff
@@ -1673,6 +1691,59 @@ static void mt_expired_timeout(struct timer_list *t)
 	clear_bit(MT_IO_FLAGS_RUNNING, &td->mt_io_flags);
 }
 
+static void update_keyboard_backlight(struct hid_device *hdev, bool enabled)
+{
+	struct usb_device *udev = hid_to_usb_dev(hdev);
+
+	unsigned char cmd_disable_backlight[] =
+		{0x24, 0xFF, 0x00, 0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+	unsigned char cmd_enable_backlight[] =
+		{0x24, 0xFF, 0x00, 0xFF, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+	unsigned char *cmd_buf;
+	unsigned char *dmabuf;
+	int ret;
+
+	/* Wake up the device in case it's already suspended */
+	if (pm_runtime_suspended(&udev->dev))
+		pm_runtime_resume(&udev->dev);
+
+	cmd_buf = enabled ? cmd_enable_backlight : cmd_disable_backlight;
+
+	dmabuf = kmemdup(cmd_buf, sizeof(cmd_buf), GFP_KERNEL);
+	if (!dmabuf) {
+		hid_err(hdev, "Failed to send type-cover hid report, couldn't allocate dmabuf\n");
+		return;
+	}
+
+	ret = hid_hw_raw_request(hdev, cmd_buf[0], dmabuf, sizeof(cmd_buf),
+				 HID_FEATURE_REPORT, HID_REQ_SET_REPORT);
```

Comment:
You might want to use `dmabuf` here exclusively, and not `cmd_buf`.

Reply:
Switched to the new `MS_TYPE_COVER_FEATURE_REPORT_SIZE`.